examples/seismic/tti/wavesolver.py

```diff
@@ -63,9 +51,15 @@ def forward(self, src=None, rec=None, u=None, v=None, m=None,
         :param u: (Optional) Symbol to store the computed wavefield
         :param m: (Optional) Symbol for the time-constant square slowness
         :param save: Option to store the entire (unrolled) wavefield
+        :param kernel: type of discretization, centered or shifted
 
         :returns: Receiver, wavefield and performance summary
         """
+
+        # Space order needs to be halved in the shifted case to have an
+        # overall space_order discretization
+        self.space_order = self.space_order / 2 if kernel == 'shifted' \
```

Comment:
should we rather to this straight away in the constructor?

Reply:
Runtime argument, the constructor can't know which kernel will be run.